go.mod

```diff
@@ -24,3 +24,11 @@ require (
 	golang.org/x/text v0.14.0 // indirect
 	gopkg.in/yaml.v3 v3.0.1 // indirect
 )
+
+godebug (
+// Allows TLS certs with negative serial numbers.
+// Before go 1.23 these certificates where accepted, now the corresponding go debug variable is needed
+// to restore the previous behavior
+// <https://cs.opensource.google/go/go/+/refs/tags/go1.23.1:src/crypto/x509/parser.go;l=1019>
+	x509negativeserial=1
+)
```

Comment:
This was detected because the integration tests became flacky [example](https://github.com/newrelic/nri-mssql/actions/runs/10828972615/job/30045513304)

```
executing: docker compose run -d --name mssql -e ACCEPT_EULA=Y -e SA_PASSWORD=secret123! -e MSSQL_PID=Developer -p1433:1433 mssql --verbose
676c222c8c30b53d62bfeaae8969b111a56b7fa89e012675349792adc2c24a0d

 Network tests_default  Creating
 Network tests_default  Created

[INFO] try to establish de connection with the mssql database...
[INFO] try to establish de connection with the mssql database...
[WARN] read tcp 127.0.0.1:44586->127.0.0.1:1433: read: connection reset by peer
[INFO] try to establish de connection with the mssql database...
[WARN] TLS Handshake failed: tls: failed to parse certificate from server: x509: negative serial number
[INFO] try to establish de connection with the mssql database...
[WARN] TLS Handshake failed: tls: failed to parse certificate from server: x509: negative serial number
[INFO] try to establish de connection with the mssql database...
[WARN] TLS Handshake failed: tls: failed to parse certificate from server: x509: negative serial number
(...)
    mssql_test.go:71: tests cannot be executed
--- FAIL: TestSuccessConnection (100.55s)
FAIL
FAIL	github.com/newrelic/nri-mssql/tests	100.554s
FAIL
 Network tests_default  Removing
 Network tests_default  Resource is still in use
make: *** [Makefile:22: integration-test] Error 1
Error: Process completed with exit code 2.
```

It can be seen as an issue with the mssql-docker image: https://github.com/microsoft/mssql-docker/issues/895
But I'd say it is better to keep the previous behior in case any user is using an invalid certifica with negative serial number (or if they'd using containerized environments based on `mssql-docker`).